test_container/tests/test/java/java_modules.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+import os
+from pathlib import Path
+import shutil
+import subprocess
+import tempfile
+import zipfile
+import requests
+from requests.auth import HTTPBasicAuth
+
+from exasol_python_test_framework import udf
+from exasol_python_test_framework import docker_db_environment
+from exasol_python_test_framework.udf import useData, expectedFailure
+from exasol_python_test_framework.udf.udf_debug import UdfDebugger
+
+script_dir = Path(os.path.dirname(os.path.realpath(__file__)))
+java_udf_dir = script_dir / "resources/java_udf"
+
+@udf.skipIfNot(docker_db_environment.is_available, reason="This test requires a docker-db environment")
+class JavaModules(udf.TestCase):
+    env = None
+    java_udf_jar_java11 = None
+    java_udf_jar_java17 = None
+    
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.java_udf_jar_java11 = self.build_java_udf(11)
+        self.java_udf_jar_java17 = self.build_java_udf(17)
+
+    def setUp(self):
+        self.query('CREATE SCHEMA JAVA_MODULES', ignore_errors=True)
+        self.query('OPEN SCHEMA JAVA_MODULES')
+        self.env = docker_db_environment.DockerDBEnvironment("JAVA_MODULE_TEST")
+
+    def build_java_udf(self, javaVersion: int) -> Path:
+        result = subprocess.run(["mvn", "--batch-mode", "--file", java_udf_dir/"pom.xml", "clean", "package", f"-Djava.version={javaVersion}"], stdout=subprocess.PIPE)
+        result.check_returncode()
+        jar = java_udf_dir / f"target/test-udf-java-{javaVersion}.jar"
+        assert jar.exists()
+        target = Path(tempfile.gettempdir()) / f"test-udf-java-{javaVersion}.jar"
+        shutil.move(jar, target)
+        return target
```

Comment:
Hm, I'am just thinking: Is it good to build the Modules for each test execution, or would it be better to build them when we build the docker image of the test container?

Reply:
@tkilias suggested to build during the test. The advantage is that changes to the Java sources will be effective immediately and don't require re-building the container.